Toshi/Source/Toshi/Render/TResource.h

```diff
@@ -5,37 +5,40 @@
 
 namespace Toshi
 {
-	class TResource : TObject, TNodeTree<TResource>::TNode
+	enum TResourceState
 	{
+		TResourceState_Valid = BITFIELD(0),
+		TResourceState_Created = BITFIELD(1),
+		TResourceState_Dying = BITFIELD(2),
+		TResourceState_Unk2 = BITFIELD(3),
+		TResourceState_Dead = BITFIELD(4),
+		TResourceState_SceneObject = BITFIELD(5),
+	};
 
+	class TResource : TObject, TNodeTree<TResource>::TNode
+	{
 	public:
+		virtual bool Create();
+		virtual bool Validate();
+		virtual void DestroyResource();
 
-		const int FLAGS_CREATED = 2;
-		const int FLAGS_DYING = 4;
-		const int FLAGS_DEAD = 16;
-		const int FLAGS_SCENEOBJECT = 32;
-		const int FLAGS_INVALID = 254;
```

Comment:
254 (FLAGS_INVALID) in binary looks like '11111110' which is actually any bits except the first one so just XOR the first bit to invalidate valid object